src/clojure/nrepl/cmdline.clj

```diff
@@ -93,16 +94,21 @@ Exit:      Control+D or (exit) or (quit)"
                      err    print
                      out    print
                      value  println}}]
-  (let [client (nrepl/client transport Long/MAX_VALUE)]
+  (let [client (nrepl/client transport Long/MAX_VALUE)
+        awaiting (atom {})]
     (println (repl-intro))
-    ;; We take 50ms to listen to any greeting messages, and display the value
-    ;; in the `:out` slot.
     (threading/run-with @threading/transport-executor
       ;; Doesn't matter which executor we hijack here, just don't use `future`.
-      (->> (client)
-           (take-while #(nil? (:id %)))
-           (run! #(when-let [msg (:out %)] (print msg)))))
-    (Thread/sleep 50)
```

Comment:
I think this is still needed. We want the greeting to be printed before the `user=>` prompt, this delay increases the chances of it happening.

Reply:
Ah, I missed that subtly.  Cool.